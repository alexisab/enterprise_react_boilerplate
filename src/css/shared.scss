/* Margins */

.mt-10 {
  margin-top: 10px;
}

.mt-20 {
  margin-top: 20px;
}

.mt-30 {
  margin-top: 20px;
}

.mt-50 {
  margin-top: 50px;
}

.mt-100 {
  margin-top: 100px;
}

.ml-8 {
  margin-left: 8px;
}

.ml-10 {
  margin-left: 10px;
}

.ml-30 {
  margin-left: 20px;
}

.ml-50 {
  margin-left: 50px;
}

.ml-100 {
  margin-left: 100px;
}

/* Dialog Form */

.dialog-form-btn-grp {
  text-align: right;
  border-top: 1px solid #e9e9e9;
  padding: 10px 16px 5px 10px;
  border-radius: 0 0 4px 4px;

  .save-btn {
    margin-left: 8px;
  }

}
