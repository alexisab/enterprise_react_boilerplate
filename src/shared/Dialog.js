import React from "react";
import PropTypes from "prop-types";
import { Modal } from "antd";

class Dialog extends React.Component {
  constructor() {
    super();
    this.state = { title: "", content: "", open: false };
  }

  open = (Component, { title }) => {
    this.setState({ open: true, content: Component, title });
  };

  close = () => {
    this.setState({ open: false, content: "", title: "" });
  };

  render() {
    const { title, content } = this.state;
    const childrenWithProps = React.Children.map(this.props.children, child =>
      React.cloneElement(child, { Dialog: this })
    );

    return (
      <div>
        {childrenWithProps}
        <Modal
          title={title}
          visible={this.state.open}
          onOk={this.close}
          onCancel={this.close}
          okText="Ok"
          cancelText="Cancel"
          footer={null}
        >
          {content}
        </Modal>
      </div>
    );
  }
}

Dialog.propTypes = {
  children: PropTypes.element.isRequired
};

export default Dialog;
