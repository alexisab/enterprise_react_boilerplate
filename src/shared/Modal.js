import React from 'react';
import { Modal } from 'antd';

class Dialog extends React.Component {
  constructor() {
    super();
    this.state = { title: '', content: '', size: '', open: false };
  }

  open = (Component, { title, size }) => {
    this.setState({ open: true, content: Component, title, size });
  }

  close = () => {
    this.setState({ open: false, content: '', title: '', size: '' });
  }

  render() {
    const { title, content } = this.state;
    const childrenWithProps = React.Children.map(this.props.children,
      child => React.cloneElement(child, { Dialog: this }),
    );

    return(
      <div>
        {childrenWithProps}
        <Modal
          title={title}
          visible={this.state.open}
          onOk={this.close}
          onCancel={this.close}
          okText="Ok"
          cancelText="Cancel"
          footer={null}
        >
          {content}
        </Modal>
      </div>
    )
  }
}

export default Dialog;
