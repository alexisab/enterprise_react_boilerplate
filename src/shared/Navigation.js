import React from 'react';
import { Link } from 'react-router-dom';
import { Layout, Menu, Icon, Avatar } from 'antd';
// import Auth from '../helpers/auth';

const { Header, Sider, Content } = Layout;
const SubMenu = Menu.SubMenu;

class Navigation extends React.Component {
  state = {
    collapsed: false,
  };
  toggle = () => {
    this.setState({
      collapsed: !this.state.collapsed,
    });
  }
  render() {
    return (
      <Layout className="main-layout">
        <Sider
          trigger={null}
          collapsible
          collapsed={this.state.collapsed}
        >
          <div className="avatar">
            <Avatar size="large" src="https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50?s=80" />
          </div>

          <Menu theme="dark" mode="inline">
            <Menu.Item key="1">
              <Link to={'/dashboard/'}>
                <Icon type="home" />
                <span>Dashboard</span>
              </Link>
            </Menu.Item>

            <Menu.Item key="2">
              <Link to={'/dashboard/admins'}>
                <Icon type="user" />
                <span>Admins</span>
              </Link>
            </Menu.Item>

            <Menu.Item key="3">
              <Link to={'/dashboard/organisations'}>
                <Icon type="rocket" />
                <span>Organisations</span>
              </Link>
            </Menu.Item>

            <SubMenu
              key="sub1"
              title={<span><Icon type="global" /><span>Menu</span></span>}
            >
              <Menu.Item key="4">Nested Menu 1</Menu.Item>
              <Menu.Item key="5">
                <Link to={'/dashboard/organisations/3pal-tax-services/businesses'}>
                  Nested Menu 2
                </Link>
              </Menu.Item>
              <Menu.Item key="6">Nested Menu 3</Menu.Item>
              <Menu.Item key="7">Nested Menu 4</Menu.Item>
            </SubMenu>

            { /*
            <Menu.Item key="8">
              <a onClick={(e) => {
                e.preventDefault();
                Auth.signOut();
                window.location.href = '/';
              }}>
                <Icon type="logout" />
                <span>Log Out</span>
              </a>
            </Menu.Item>
            */ }

          </Menu>
        </Sider>
        <Layout>
          <Header style={{ background: '#fff', padding: 0 }}>
            <Icon
              className="trigger"
              type={this.state.collapsed ? 'menu-unfold' : 'menu-fold'}
              onClick={this.toggle}
            />
          </Header>
          <Content style={{ margin: '24px 16px', padding: 24, background: '#fff', minHeight: 280 }}>
            {this.props.children}
          </Content>
        </Layout>
      </Layout>
    );
  }
}

export default Navigation;
