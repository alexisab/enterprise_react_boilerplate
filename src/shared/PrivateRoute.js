import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { verifyToken } from '../helpers/auth';
import { Loader } from '../helpers/loader';
import Dialog from './Dialog';
import Navigation from './Navigation';

class PrivateRoute extends React.Component {
  state = {
    loading: true,
    currentUser: null
  }

  componentDidMount() {
    console.log('authenticating...');

    verifyToken().then(({ data }) => {
      this.setState({
        loading: false,
        currentUser: data.verifyToken,
      });
    });
  }

  render() {
    const { component: Component, ...rest } = this.props;
    const { currentUser, loading } = this.state;
    if (loading) {
      return (<Loader />);
    }

    return (
      <Route
        {...rest}
        render={props => (
          <div>
            { currentUser === null && <Redirect to={{ pathname: '/', state: { from: this.props.location } }} />}
            <Navigation currentUser={currentUser}>
              <Dialog>
                <Component {...props} currentUser={currentUser} />
              </Dialog>
            </Navigation>
          </div>
        )}
      />
    );
  }
}

export default PrivateRoute;
