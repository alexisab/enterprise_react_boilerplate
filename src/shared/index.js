import Navigation from './Navigation';
import Loader from './Loader';

export { Navigation, Loader }
