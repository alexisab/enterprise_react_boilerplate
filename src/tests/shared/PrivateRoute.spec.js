import React from "react";
import renderer from "react-test-renderer";
import { PrivateRoute } from "../../shared";

describe("PrivateRoute", () => {
  test("it should match snapshot", () => {
    const component = renderer.create(
      <PrivateRoute path="/dashboard" component={() => <div>Dummy</div>} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
