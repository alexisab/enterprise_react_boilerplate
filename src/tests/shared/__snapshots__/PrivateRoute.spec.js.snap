// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute it should match snapshot 1`] = `
<div>
  Loading...
</div>
`;
